services/src/main/java/org/keycloak/authentication/authenticators/browser/SpnegoAuthenticator.java

```diff
@@ -86,7 +86,7 @@ public void authenticate(AuthenticationFlowContext context) {
         String spnegoToken = tokens[1];
         UserCredentialModel spnegoCredential = UserCredentialModel.kerberos(spnegoToken);
 
-        CredentialValidationOutput output = context.getSession().userCredentialManager().authenticate(context.getSession(), context.getRealm(), spnegoCredential);
+        CredentialValidationOutput output = context.getSession().users().getUserByCredential(context.getRealm(), spnegoCredential);
```

Comment:
Reads weird to me because the code is actually validating credentials and not only querying a credential store. I would say that the previous code makes more sense where the `authenticate` method is called. Perhaps `validate` or `isValid` makes sense because we are actually validating a credential.